src/moscot/problems/cross_modality/_mixins.py

```diff
@@ -0,0 +1,173 @@
+    Any,
+    Dict,
+    List,
+    Literal,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+)
+
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+import scipy.spatial
+import anndata as ad
+import scanpy as sc
+
+
+from moscot import _constants
+from moscot._docs._docs import d
+from moscot._docs._docs_mixins import d_mixins
+from moscot._logging import logger
+from moscot._types import ArrayLike, Device_t, Str_Dict_t
+from moscot.base.problems._mixins import AnalysisMixin, AnalysisMixinProtocol
+from moscot.base.problems.compound_problem import B, K
+from moscot.utils.subset_policy import StarPolicy
+from moscot.base.output import BaseSolverOutput
+
+
+__all__ = ["CrossModalityTranslationMixin"]
+
+class CrossModalityTranslationMixinProtocol(AnalysisMixinProtocol[K, B]):
+    """Protocol class."""
+
+    adata_src: AnnData
+    adata_tgt: AnnData
+    src_attr: Optional[str]
+    tgt_attr: Optional[str]
+
+    def _cell_transition(
+            self: AnalysisMixinProtocol[K, B], 
+            *args: Any, 
+            **kwargs: Any
+    ) -> pd.DataFrame:
+        ...
+
+class CrossModalityTranslationMixin(AnalysisMixin[K, B]):
+    """Cross modality translation analysis mixin class."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._src_attr: Optional[str] = None
+        self._tgt_attr: Optional[str] = None
+
+    def translate(
+            self: CrossModalityTranslationMixinProtocol[K, B],
+            source: K,
+            target: K,
+            forward: bool = True,
+            **kwargs: Any,
+    ) -> ArrayLike:
+        """
+        Translate source or target object.
+
+        Parameters
+        ----------
+        source
+            Key identifying the source distribution.
+        target
+            Key identifying the target distribution.
+        forward
+            If `True` computes translation from source object to target object, otherwise backward.
+        kwargs
+            keyword arguments for policy-specific `_apply` method of :class:`moscot.problems.CompoundProblem`.
+        
+        """
+        if forward:
+            self._translation = self[(source, target)].solution.pull(self.adata_tgt.obsm[self._tgt_attr], scale_by_marginals=True, **kwargs)
+            # _translation_alternative = self.problems[source, target].pull(self.adata_tgt.obsm[self._tgt_attr], scale_by_marginals=True, **kwargs) 
+            # Note: to use the alternative, I had to comment out the negative mass error
+        else:
+            if self.batch_key is None:
+                self._translation = self[(source, target)].solution.push(self.adata_src.obsm[self._src_attr], scale_by_marginals=True, **kwargs)   
+            else:
+                self._translation = self[(source, target)].solution.push(self.adata_src[self.adata_src.obs[self.batch_key]==source].obsm[self._src_attr], scale_by_marginals=True, **kwargs)       
```

Comment:
`self[source, target].adata_src.obsm[self._src_attr]` and access push/pull of CompoundProblem(AnalysisMixin)